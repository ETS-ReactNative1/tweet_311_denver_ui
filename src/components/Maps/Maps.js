import React, { Component } from 'react';
import { StyleSheet, Dimensions, Text, View, TouchableOpacity, Image } from 'react-native';
import MapView, { Marker } from 'react-native-maps';

const height = Dimensions.get('window').height;
const width = Dimensions.get('window').width;

class Maps extends Component {
  constructor(props) {
    super(props);
    this.state = {
      markers: []
    }
  }

  handleValidation = () => {
    const { markers } = this.state;
    const greenCheck = <Image style={styles.greenCheck} source={require('../../../assets/images/green_check.png')}></Image>
    const redX = <Image style={styles.redCheck} source={require('../../../assets/images/stop.png')}></Image>

    if(!markers.length) {
      return(
        <View style={styles.xContainer}>
          <Text style={styles.redText}>Please Select A Location</Text>
          <Text>{redX}</Text>
        </View>
      )
    } else {
      return(
        <Text>{greenCheck}</Text>
      )
    }
  }

  handlePress = (e) => {
    this.setState({
      markers: [
        {
          coordinate: e.nativeEvent.coordinate,
          pin: <Image style={styles.imgPin} source={require('../../../assets/images/pin.png')} />
        }
      ]
    });
  };

  handleSubmit = () => {
    const { navigation, setLocation } = this.props;
    const { markers } = this.state;

    if(markers.length) {
      const latAndLong = {
        lat: markers[0].coordinate.latitude,
        long: markers[0].coordinate.longitude
      };
      setLocation(latAndLong.lat, latAndLong.long);
      navigation.navigate('Home');
    } else {
      navigation.navigate('Home');
    }
  };

  getLocation = () => {
    const { markers } = this.state;

    return markers.map(marker => {
      return (
        <Marker { ...marker } key={marker.coordinate}>
          <View>
            <Text style={styles.img}>{marker.pin}</Text>
          </View>
        </Marker>)
    })
  };

  render() {
    const { markers } = this.state;

    return(
      <View style={styles.mapContainer}>
        <MapView
        style={styles.map}
        region={ {
          latitude: 39.7392,
          longitude: -104.9903,
          latitudeDelta: 0.0922,
          longitudeDelta: 0.0421,
        }
      }
        showsTraffic={true}
        onPress={ this.handlePress }
        >
        {!markers.length ? null : this.getLocation()}
        </MapView>
        {this.handleValidation()}
        <TouchableOpacity style={styles.btn} onPress={ () => this.handleSubmit() }>
          <Text style={styles.text}>Add Location</Text>
        </TouchableOpacity>
      </View>
    )
  }
};

const styles = StyleSheet.create({
  mapContainer: {
    alignItems: 'center',
    backgroundColor: '#fff',
    flexDirection: 'column',
    flex: 1,
    width: '100%',
  },
  map: {
    borderColor: '#3976EA',
    borderWidth: 1,
    height: height * .65,
    width: '100%',
  },
  xContainer: {
    alignItems: 'center',
    flexDirection: 'column',
    flex: 1,
  },
  text: {
    color: '#FFFFFF',
    fontSize: 20
  },
  redCheck: {
    marginLeft: 'auto',
    marginRight: 'auto',
    height: 35,
    width: 35,
  },
  greenCheck: {
    height: 45,
    marginTop: 25,
    width: 60,
  },
  redText: {
    color: 'red',
    fontSize: 15,
    marginTop: height * .04,
    marginBottom: height * .02,
  },
  imgPin: {
    height: 30,
    width: 30,
  },
  btn: {
    alignItems: 'center',
    backgroundColor: '#3976EA',
    borderRadius: 20,
    height: 45,
    justifyContent: 'center',
    marginBottom: height * .06,
    marginLeft: 'auto',
    marginRight: 'auto',
    marginTop: height * .04,
    width: '50%'
  }
});

export default Maps;
