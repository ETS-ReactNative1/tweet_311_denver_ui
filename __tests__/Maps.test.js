import React from 'react';
import renderer from 'react-test-renderer';
import MockMapView from 'react-native-maps';

import Maps from '../src/components/Maps/Maps';

describe('Maps', () => {

  test('renders Maps correctly', () => {
    const tree = renderer.create(<Maps />).toJSON();
    expect(tree).toMatchSnapshot();
  });

});



