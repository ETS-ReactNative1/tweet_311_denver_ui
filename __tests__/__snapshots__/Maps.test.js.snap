// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Maps renders Maps correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "flexDirection": "column",
      "width": "100%",
    }
  }
>
  <View />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "red",
          "fontSize": 18,
          "marginBottom": 26.68,
          "marginTop": 26.68,
        }
      }
    >
      Please Select A Location
    </Text>
    <Text>
      <Image
        source={1}
        style={
          Object {
            "height": 40,
            "marginLeft": "auto",
            "marginRight": "auto",
            "width": 40,
          }
        }
      />
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#3976EA",
        "borderRadius": 20,
        "height": 45,
        "justifyContent": "center",
        "marginBottom": 80.03999999999999,
        "marginLeft": "auto",
        "marginRight": "auto",
        "marginTop": 53.36,
        "opacity": 1,
        "width": 375,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      Add Location
    </Text>
  </View>
</View>
`;
